Ryujinx.HLE/HOS/Services/Vi/IApplicationDisplayService.cs

```
@@ -204,7 +211,7 @@ public long ConvertScalingMode(ServiceCtx context)
                 case SrcScalingMode.PreserveAspectRatio: return DstScalingMode.PreserveAspectRatio;
             }
 
-            return null;
+            throw new ArgumentException($"Invalid scaling mode \"{source}\" specified.");
```

Should we throw here, or is there some sort of "Generic Error" code we can return from ConvertScalingMode?

---

Invalid enum values (> 4) returns error code 0x272 on Vi. Maybe we could return the same error code here? It's also possible to implement it as a `Try*` method, return a bool and a out parameter with the converted scaling mode.

---

I avoided that (out param) in my initial implementation because it made the switch case messy to implement (can't cleanly just return from each case anymore). My initial impl got around this by using a nullable return type, and catching a null in the caller and returning an error message. Perhaps we could do this?

I guess the question is, do we prefer to assert this in the host, or in the guest? I'd imagine returning 0x272 in the guest will cause it to assert?

---

Imo returning the error code is the right thing to do. This exception will never be hit, because it checks whenever or not the enum value is valid beforehand.

---

Then I think we should return the error code over throwing an exception. I prefer my initial implementation (returning a nullable) over using an out parameter though, but up to you.